database/mysql.go

```diff
@@ -191,29 +245,55 @@ func RunMySQLDockerInstance(config *DatabaseConfig) (*DatabaseConfig, *dockertes
 		config.MySQL.Password = "secret"
 	}
 
-	pool, err := dockertest.NewPool("")
+	resource, err := findOrLaunchMySQLContainer(config)
 	if err != nil {
 		return nil, nil, err
 	}
+	address := fmt.Sprintf("tcp(localhost:%s)", resource.GetPort("3306/tcp"))
 
-	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
-		Repository: "mysql",
-		Tag:        "8",
+	return &DatabaseConfig{
+		DatabaseName: config.DatabaseName,
+		MySQL: &MySQLConfig{
+			Address:  address,
+			User:     config.MySQL.User,
+			Password: config.MySQL.Password,
+		},
+	}, resource, nil
+}
+
+func findOrLaunchMySQLContainer(config *DatabaseConfig) (*dockertest.Resource, error) {
+	var containerName = "mysql-test-container"
+	var resource *dockertest.Resource
+	var err error
+
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = pool.RunWithOptions(&dockertest.RunOptions{
+		Name:       containerName,
+		Repository: "vaulty/mysql-volumeless",
+		Tag:        "8.0",
 		Env: []string{
 			fmt.Sprintf("MYSQL_USER=%s", config.MySQL.User),
 			fmt.Sprintf("MYSQL_PASSWORD=%s", config.MySQL.Password),
-			"MYSQL_ROOT_PASSWORD=secret",
-			fmt.Sprintf("MYSQL_DATABASE=%s", config.DatabaseName),
+			fmt.Sprintf("MYSQL_ROOT_PASSWORD=%s", config.MySQL.Password),
 		},
-	}, func(dockerConfig *dc.HostConfig) {
-		dockerConfig.AutoRemove = true
-	},
-	)
-	if err != nil {
-		return nil, nil, err
+	})
+
+	if err != nil && !errors.Is(err, dc.ErrContainerAlreadyExists) {
+		return nil, err
+	}
+
+	// look for running container
+	resource, found := pool.ContainerByName(containerName)
```

Comment:
This should be at the beginning of this function:
```go
resource, ok := pool.ContainerByName(containerName)
if ok { return resource, nil }
```

Reply:
I did it intentionally. There maybe no container first. When we create container there maybe already one :)

That’s why it tries to create container first, ignore if it exists and only after that look for container by name.